### ISSUE TITLE
convert `args` to `kwargs` -> reduce keys for identical inputs

### ISSUE THREAD

**Issue by Borda** [python-cachier/cachier#138]
It seems that calling the same sample function with identical argument values can yield two different entry keys; see an example of a simple function:
```py
def sample_keys(a, b, c=3):
    return a + b + c
```
The default hash function 
https://github.com/python-cachier/cachier/blob/28c8d0d6beece98e64eb4af7a013fe6562c87776/cachier/core.py#L83-L89
will give these keys:
- for `sample_keys(1, 2, c=3)` gives:
	```py
	args, kwds = (1, 2), {"c": 3}
	functools._make_key(args, kwds, typed=True)
	Out[1]: [1, 2, <object at 0x7fc500943380>, 'c', 3, int, int, int]
	```
- for `sample_keys(1, b=2, c=3)` gives:
	```py
	args, kwds = (1,), {"b": 2, "c": 3}
	functools._make_key(args, kwds, typed=True)
	Out[2]: [1, <object at 0x7fc500943380>, 'b', 2, 'c', 3, int, int, int]
	```

**Comment by Borda** [comment]
So my suggestion is to immediately convert all `args` to `kwargs`, for example: https://stackoverflow.com/a/831164/4521646

**Comment by shaypal5** [comment]
Looks good! If you're willing to implement it, I'd suggest giving it a go; If all tests pass locally, I'll pull it into a dev branch to run all Mongo core tests against an actual MongoDB server (I just can't get PRs to run tests with secrets), and then - if it doesn't break anything - it can be merged and released. 🥇

**Comment by Borda** [comment]
fixed in #140 and improved in #134
